### ISSUE TITLE
Opencart4 not working - /install/install/install/.../index.php redirect loop

### ISSUE THREAD

**Issue by MrVilko** [bitnami/containers#22083]
### Name and Version

bitnami/opencart:4.0.1-1-debian-11-r47

### What steps will reproduce the bug?

I have a couple of issues, that might (or not) be connected:

1. Container cannot be restarted. After a restart, it won't come back up. I had "a bunch" of different errors that I had not recorded. The last one is:
```
opencart 23:43:03.19 INFO  ==> Trying to connect to the database server
[27-Jan-2023 23:43:03 UTC] PHP Fatal error:  Uncaught Error: Undefined constant "DB_HOSTNAME" in Command line code:1
Stack trace:
#0 {main}
  thrown in Command line code on line 1
```
To make it work again I have to remove the container as well as any persisted data.
DB connection works on the first startup. I tried providing a bad hostname and correctly got an error. If I provide the correct one, I get to apache logs and see access attempts.

2. However, it looks like this:
```
opencart 23:34:50.13 INFO  ==> ** Starting Apache **
[Fri Jan 27 23:34:50.157508 2023] [ssl:warn] [pid 1] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Fri Jan 27 23:34:50.157698 2023] [ssl:warn] [pid 1] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Fri Jan 27 23:34:50.170344 2023] [ssl:warn] [pid 1] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Fri Jan 27 23:34:50.170508 2023] [ssl:warn] [pid 1] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Fri Jan 27 23:34:50.179034 2023] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.55 (Unix) OpenSSL/1.1.1n configured -- resuming normal operations
[Fri Jan 27 23:34:50.179053 2023] [core:notice] [pid 1] AH00094: Command line: '/opt/bitnami/apache/bin/httpd -f /opt/bitnami/apache/conf/httpd.conf -D FOREGROUND'
172.18.0.1 - - [27/Jan/2023:23:34:59 +0000] "GET / HTTP/1.1" 302 134
172.18.0.1 - - [27/Jan/2023:23:34:59 +0000] "GET /install/index.php HTTP/1.1" 302 134
172.18.0.1 - - [27/Jan/2023:23:34:59 +0000] "GET /install/install/index.php HTTP/1.1" 302 134
172.18.0.1 - - [27/Jan/2023:23:34:59 +0000] "GET /install/install/install/index.php HTTP/1.1" 302 134
172.18.0.1 - - [27/Jan/2023:23:34:59 +0000] "GET /install/install/install/install/index.php HTTP/1.1" 302 134
172.18.0.1 - - [27/Jan/2023:23:34:59 +0000] "GET /install/install/install/install/install/index.php HTTP/1.1" 302 134
172.18.0.1 - - [27/Jan/2023:23:34:59 +0000] "GET /install/install/install/install/install/install/index.php HTTP/1.1" 302 134
172.18.0.1 - - [27/Jan/2023:23:34:59 +0000] "GET /install/install/install/install/install/install/install/index.php HTTP/1.1" 302 134
```
My docker-compose:
```yaml
version: '2'
services:
  mariadb:
    image: docker.io/bitnami/mariadb:10.6
    user: root
    container_name: mariadb
    environment:
      - MARIADB_ROOT_PASSWORD=-------------------
      - MARIADB_USER=bn_opencart
      - MARIADB_PASSWORD=----------------------
      - MARIADB_DATABASE=bitnami_opencart
    volumes:
      - ./mariadb/data:/bitnami/mariadb/data
      - ./mariadb/logs:/opt/bitnami/mariadb/logs
    networks:
      - opencart
  opencart:
    image: docker.io/bitnami/opencart:4
    ports:
      - '9080:80'
      - '9443:443'
    user: root
    container_name: opencart
    environment:
      - OPENCART_HOST=example.domain
      - OPENCART_DATABASE_HOST=mariadb
      - OPENCART_DATABASE_PORT_NUMBER=3306
      - OPENCART_DATABASE_USER=bn_opencart
      - OPENCART_DATABASE_PASSWORD=--------------------------
      - OPENCART_DATABASE_NAME=bitnami_opencart
      - OPENCART_ENABLE_HTTPS=on
      - APACHE_HTTP_PORT_NUMBER=80
      - APACHE_HTTPS_PORT_NUMBER=443
      - OPENCART_HOST=admin
      - OPENCART_PASSWORD=----------------
    volumes:
      - ./opencart/data:/bitnami/opencart
      - ./opencart/storage_data:/bitnami/opencart_storage/
    networks:
      - opencart
    depends_on:
      - resinosopencart-mariadb
```

I tried with https on/off. Different OPENCART_HOSTS (domain, localhost, container hostname).
I set `user: root` to overcome permission issues on volumes, although docker runs under nonroot user. This seemed to be the fastest way to start the container up.

### What is the expected behavior?

Opencart "should work" without a redirect loop and "The page isn’t redirecting properly" error in the browser.

### What do you see instead?

The page isn’t redirecting properly

Firefox has detected that the server is redirecting the request for this address in a way that will never be complete.

**Comment by carrodher** [comment]
Please, [note that the Bitnami team is not actively monitoring GitHub Issues for this project](https://github.com/bitnami/containers/tree/main/bitnami/opencart#community-supported-solution). This issue will stay open for 20 days to allow the community to contribute, after 15 days without activity the issue will be marked as stale, being closed after 5 days.

If you would like to contribute to this container image by fixing a bug or adding a new feature, feel free to create a PR, the Bitnami team will be happy to review it and provide feedback.

**Comment by zen** [comment]
@MrVilko I have the same issue with very basic configuration similar to your) DId you ever solve this?

**Comment by MrVilko** [comment]
Hi @zen. I ended up making changes to the Opencart code in the container.

First of all, the db error from the log was unrelated and probably caused by me trying a bunch of wild stuff...

Now to the redirection... From what I remember, `OPENCART_HOST` wasn't being set properly in Opencart config. I think `HTTP_SERVER` in both config.php and administration/config.php was correctly `http://example.domain` but `HTTPS_SERVER` was being set to `http://example.domain` as well (instead of `https://example.domain`).

If I remember well, even when set correctly there were some redirect issues so I ended up rewriting both `HTTP_SERVER` and `HTTPS_SERVER` to `https://example.domain` by hand (which opens a whole new can of worms). 

I know I did a bunch of other changes but I think they were mainly cosmetic...
